calico_containers/adapter/datastore.py

```diff
@@ -624,12 +669,9 @@ def get_profile_members(self, profile_name):
             endpoints = self.etcd_client.read(ALL_ENDPOINTS_PATH,
                                               recursive=True).leaves
             for child in endpoints:
-                packed = child.key.split("/")
-                if len(packed) == 10:
-                    (_, _, _, _, host, _, ctype, cid, _, ep_id) = packed
-                    ep = Endpoint.from_json(ep_id, child.value)
-                    if ep.profile_id == profile_name:
-                        eps[host][ctype][cid][ep_id] = ep
+                ep = Endpoint.from_json(child.key, child.value)
+                if ep and profile_name in ep.profile_ids:
+                    eps[ep.hostname][ep.orchestrator_id][ep.workload_id][ep.endpoint_id] = ep
```

Comment:
Would be nice for this to just return a list rather than a nested dictionary and remote the get_profile_members_endpoint_ids method.  However - I don't suggest we do that just yet.  I'll create a google doc to keep track of future improvements.